job-runner/runner/runner.go

```diff
@@ -128,22 +131,30 @@ TRY_LOOP:
 			errMsg = jobRet.Error.Error()
 		}
 
+		// Can be stopped while running, in which case STATE_FAIL is not really
+		// because it failed but because we stopped it, so log then overwrite
+		// the state = stopped. This also sets finalState below.
+		if r.stopped() {
+			if jobRet.State != proto.STATE_STOPPED && jobRet.State != proto.STATE_COMPLETE {
+				tryLogger.Errorf("job stoped: changing state %s (%d) to STATE_STOPPED", proto.StateName[jobRet.State], jobRet.State)
+				jobRet.State = proto.STATE_STOPPED
+			}
+		}
+
 		// Create a JL and send it to the RM.
 		jl := proto.JobLog{
-			RequestId:   r.reqId,
-			JobId:       r.jobId,
-			Name:        r.jobName,
-			Type:        r.jobType,
-			Try:         r.prevTryNo + tryNo,
-			SequenceId:  r.sequenceId,
-			SequenceTry: r.sequenceTry,
-			StartedAt:   startedAt,
-			FinishedAt:  finishedAt,
-			State:       jobRet.State,
-			Exit:        jobRet.Exit,
-			Error:       errMsg,
-			Stdout:      jobRet.Stdout,
-			Stderr:      jobRet.Stderr,
+			RequestId:  r.reqId,
+			JobId:      r.jobId,
+			Name:       r.jobName,
+			Type:       r.jobType,
+			Try:        r.totalTries + tryNo, // job_log.try is monotonically increasing across all seq tries
```

Comment:
Won't this result in some weirdness (skipped Try numbers) for resumed jobs? tryNo = tries before stop + tries by this runner, but totalTries already includes those tries before being stopped.

Reply:
I think you can do `r.totalTries + tryNo - r.tries`

Reply:
Yes, fixed by using `totalTries` instead, and incrementing it in the loop, too. Will write a test case, too.